op-sync-tester/synctester/backend/sync_tester.go

```diff
@@ -0,0 +1,153 @@
+package backend
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+	"sync"
+
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum-optimism/optimism/op-sync-tester/metrics"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/rpc"
+
+	"github.com/ethereum-optimism/optimism/op-sync-tester/synctester/backend/config"
+	sttypes "github.com/ethereum-optimism/optimism/op-sync-tester/synctester/backend/types"
+	"github.com/ethereum-optimism/optimism/op-sync-tester/synctester/frontend"
+)
+
+var (
+	ErrNoSession  = errors.New("no session")
+	ErrNoReceipts = errors.New("no receipts")
+)
+
+type SyncTester struct {
+	mu sync.RWMutex
+
+	log log.Logger
+	m   metrics.Metricer
+
+	id       sttypes.SyncTesterID
+	chainID  eth.ChainID
+	elClient *ethclient.Client
```

Comment:
Food for thought, should we both initialize `ethclient.Client` and `sources.Client(or `client.RPC`) for querying the backend EL? Each have different abstractions and have pros and cons.